Userland/Libraries/LibAudio/PulseAudioOutputImplementation.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, Gregory Bertilson <zaggy1024@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PulseAudioOutputImplementation.h"
+
+namespace Audio {
+
+ErrorOr<NonnullOwnPtr<OutputImplementation>> PulseAudioOutputImplementation::create(OutputState initial_state, u32 sample_rate, u8 channels, u32 target_latency_ms, AudioDataRequestCallback const& data_request_callback)
+{
+    VERIFY(data_request_callback);
+    auto context = TRY(PulseAudioContext::instance());
+    auto stream = TRY(context->create_playback_stream(initial_state, sample_rate, channels, target_latency_ms, [&](PulseAudioPlaybackStream&, Bytes buffer, size_t sample_count) {
+        VERIFY(data_request_callback);
+        return data_request_callback(buffer, PcmSampleFormat::Float32, sample_count);
```

Comment:
Is pulse audio always floating-point? IIRC on my Linux box, Qt reported that my audio wanted 16-bit ints by default, whereas my mac wanted floats. Using the desired sample type is one of the things that made audio sound better on mac, see:
https://github.com/SerenityOS/serenity/commit/afe53a4856e232ae72e020cfe5140ac75788769c#diff-519b54f84f65c64c7d98072ef73a9907ed5df7f50f73978f7a6ca86201ed18a8R188

Reply:
Did this affect things on Linux running PulseAudio? I'd personally rather leave the conversion up to the server if possible, since it will also be doing resampling itself. We could certainly test whether it makes a difference, but I haven't noticed anything suspicious yet.

Reply:
Not sure tbh! It might've just been mac that was weird, or something else that was fixed in that commit. FWIW audio sounds fine on my desktop with floats, so this could certainly be a non-issue.

Reply:
Yeah, depending on what you were hearing it seems like it definitely could've been a problem with not writing the correct amount of data to the buffers. We can keep an ear out for anything suspicious anyway, I'll resolve this.